vendor/visualmetrics.py

```diff
@@ -1457,6 +1462,41 @@ def calculate_speed_index(progress):
         last_progress = p['progress'] / 100.0
     return int(si)
 
+def calculate_contentful_speed_index(progress, directory):
+    x = len(progress)
```

Comment:
It looks like `x` is used only once, and then just for indexing the last element.  You can use `progress[-1]` for that in Python.